navbar.ts

```diff
@@ -0,0 +1,433 @@
+import { LitElement } from "lit";
+import { customElement, property } from "lit/decorators.js";
+import "./accordion";
+
+interface NodeType {
+  elementName: string;
+  children: NodeType[];
+}
+@customElement("aalam-navbar")
+export class AalamNavbar extends LitElement {
+  @property({ type: String }) attr = "";
+  @property({ type: String }) mode = "saccordion";
+  @property({ type: Number }) cutoffwidth = 0;
+  @property({ type: String }) togglesel = "nav-menu";
+  @property({ type: String }) pos = "right";
+  @property({ type: String }) canvascls = "canvas-body";
+
+  noParentAnchors: NodeType[] = [];
+  noParentContianer: HTMLAnchorElement[] = [];
+  parentContainer: { [key: string]: HTMLAnchorElement[] } = {};
+  anchorElements: HTMLAnchorElement[];
+  menu: HTMLElement;
+  menuElement: HTMLAnchorElement;
+  closeElement: HTMLAnchorElement;
+  backElement: HTMLAnchorElement;
+  currentPath: HTMLAnchorElement[] = [];
+  change: boolean = false;
+  overlayContainer: { [key: string]: HTMLElement[] } = {};
+  activeClickOutsideListener: ((event: MouseEvent) => void) | null = null;
+  navbar: HTMLElement;
+
+  override createRenderRoot() {
+    return this;
+  }
+  override firstUpdated() {
+    this.navbar = document.querySelector("aalam-navbar") as HTMLElement;
+
+    const original = this.querySelectorAll(
+      this.attr ? `[${this.attr}]` : "a"
+    ) as NodeListOf<HTMLAnchorElement>;
+    this.anchorElements = Array.from(original).map(
+      (anchor) => anchor.cloneNode(true) as HTMLAnchorElement
+    );
+
+    const lengthOfInput = this.anchorElements.length;
+
+    let i = 0;
+    while (i < lengthOfInput) {
+      const anchor = this.anchorElements[i];
+      const attrValue = anchor.getAttribute(this.attr) || "";
+      const attributes = this.parseAttributes(attrValue);
+      const attrName = attributes.nm;
+
+      if (!attributes.pnt) {
+        this.noParentAnchors.push({ elementName: attrName, children: [] });
+        if (!(attrName === "menu") && !(attrName == "back"))
+          this.noParentContianer.push(anchor);
+      } else {
+        if (!this.parentContainer[attributes.pnt]) {
+          this.parentContainer[attributes.pnt] = [];
+        }
+        this.parentContainer[attributes.pnt].push(anchor);
+      }
+      i++;
+    }
+    
+    const toggle = this.querySelector(this.togglesel);
+    if (toggle) {
+      const originalMenu = toggle as HTMLAnchorElement;
+      this.menuElement = originalMenu.cloneNode(true) as HTMLAnchorElement;
+      originalMenu.style.display = "none";
+      document.body.appendChild(this.menuElement);

```

Comment:
No need to clone the menu element, instead we can keep it assigned in a variable and set the style.display to none when the screen size is above the cutoff width and style.display to "" when the screen size is below the cutoff width